plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/element/ApplyAclRuleVspCommand.java

```diff
@@ -79,4 +191,47 @@ public boolean executeInSequence() {
         return false;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        ApplyAclRuleVspCommand that = (ApplyAclRuleVspCommand) o;
+
+        if (_egressDefaultPolicy != that._egressDefaultPolicy) return false;
+        if (_isL2Network != that._isL2Network) return false;
+        if (_networkAcl != that._networkAcl) return false;
+        if (_networkId != that._networkId) return false;
+        if (_networkReset != that._networkReset) return false;
+        if (_aclRules != null ? !_aclRules.equals(that._aclRules) : that._aclRules != null) return false;
+        if (_acsIngressAcl != null ? !_acsIngressAcl.equals(that._acsIngressAcl) : that._acsIngressAcl != null)
+            return false;
+        if (_domainTemplateName != null ? !_domainTemplateName.equals(that._domainTemplateName) : that._domainTemplateName != null)
+            return false;
+        if (_networkDomainUuid != null ? !_networkDomainUuid.equals(that._networkDomainUuid) : that._networkDomainUuid != null)
+            return false;
+        if (_networkName != null ? !_networkName.equals(that._networkName) : that._networkName != null) return false;
+        if (_networkUuid != null ? !_networkUuid.equals(that._networkUuid) : that._networkUuid != null) return false;
+        if (_vpcOrSubnetUuid != null ? !_vpcOrSubnetUuid.equals(that._vpcOrSubnetUuid) : that._vpcOrSubnetUuid != null)
+            return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = (_networkAcl ? 1 : 0);
+        result = 31 * result + (_networkUuid != null ? _networkUuid.hashCode() : 0);
+        result = 31 * result + (_networkDomainUuid != null ? _networkDomainUuid.hashCode() : 0);
+        result = 31 * result + (_vpcOrSubnetUuid != null ? _vpcOrSubnetUuid.hashCode() : 0);
+        result = 31 * result + (_networkName != null ? _networkName.hashCode() : 0);
+        result = 31 * result + (_isL2Network ? 1 : 0);
+        result = 31 * result + (_aclRules != null ? _aclRules.hashCode() : 0);
+        result = 31 * result + (int) (_networkId ^ (_networkId >>> 32));
+        result = 31 * result + (_egressDefaultPolicy ? 1 : 0);
+        result = 31 * result + (_acsIngressAcl != null ? _acsIngressAcl.hashCode() : 0);
+        result = 31 * result + (_networkReset ? 1 : 0);
+        result = 31 * result + (_domainTemplateName != null ? _domainTemplateName.hashCode() : 0);
+        return result;
+    }
```

Comment:
Please implement ``toString()`` to provide debugging state information for debugging purposes.